wikimedia/wikipedia-ios
# Article as a Living Doc Model - enable thanks button

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T259555

### Notes
* This refactors the thanking flow from the `DiffContainerViewController` to a new protocol and protocol extension named... wait for it... `ThanksGiving`. 🦃 
* I don't know why it won't work without that `reloadData` call. That breaks my understanding of diffable data sources.

### Test Steps
On both an edit history diff (to ensure no regressions) and the AALD modal:
1. Start not logged in. Try to tap a greyed out "Thank" button. Ensure you get a nice message, and can log in. Once you log in, the button should change colors.
1. Tap the button to give thanks. Ensure you get proper feedback. Try to tap it again. Ensure you get proper feedback.
1. Try to thank an anonymous user. Ensure you get proper feedback.

### Screenshots/Videos

<img width="300" alt="Screenshot 4" src="https://user-images.githubusercontent.com/9295855/95528044-e14dab80-098b-11eb-8480-9ec9479d7d40.png"> <img width="200" alt="Screenshot 2" src="https://user-images.githubusercontent.com/9295855/95528047-e3176f00-098b-11eb-987b-aa1e7974ef75.png"> <img width="220" alt="Screenshot 3" src="https://user-images.githubusercontent.com/9295855/95528049-e4e13280-098b-11eb-86e7-ee62c1804538.png">
